pkg/virt-controller/watch/vm.go

```diff
@@ -493,6 +493,43 @@ func (c *VMController) arePVCVolumesReady(vm *virtv1.VirtualMachine) bool {
 	return true
 }
 
+func (c *VMController) hasDataVolumeErrors(vm *virtv1.VirtualMachine) bool {
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.DataVolume == nil {
+			continue
+		}
+
+		dvKey := fmt.Sprintf("%s/%s", vm.Namespace, volume.DataVolume.Name)
+		dvObj, exists, err := c.dataVolumeInformer.GetStore().GetByKey(dvKey)
+		if err != nil {
+			log.Log.Object(vm).Errorf("Error fetching DataVolume %s: %v", dvKey, err)
+			continue
+		}
+		if !exists {
+			log.Log.Object(vm).Warningf("DataVolume %s referenced by VM %s/%s does not exists",
+				dvKey, vm.Namespace, vm.Name)
+			continue
+		}
+
+		dv := dvObj.(*cdiv1.DataVolume)
+
+		if dv.Status.Phase == cdiv1.Failed {
+			log.Log.Object(vm).Errorf("DataVolume %s is in Failed phase", dvKey)
+			return true
```

Comment:
We could potentially differentiate the Failed from Error, since Error means that it will retry, while failed means that it will not, I guess? But I am not sure if it is necessary, because the difference is probably not wide enough known to justify the difference in the printable status field.

Reply:
I agree that the difference (if any) isn't significant nor evident enough to merit a different status value. Either way the printableStatus is more of a first hint to the user that something is not working with a VM, after which he should really look at the VM's DVs to get more info on the actual error.